### PR TITLE
Esc for Regular Expression on Rewrite Terms

### PR DESCRIPTION
**Pull request by bordoni**
_Ref:_ [C#42789](https://central.tri.be/issues/42789)

### CODE REVIEW

File: src/Tribe/Rewrite.php [hunk 1]
```diff
@@ -250,7 +250,7 @@ public function get_bases( $method = 'regex' ) {
 				'today' => array( 'today', $tec->todaySlug ),
 				'day' => array( 'day', $tec->daySlug ),
 				'tag' => array( 'tag', $tec->tag_slug ),
-				'tax' => array( 'category', $tec->category_slug ),
+				'tax' => array( 'category-testing', $tec->category_slug ),
```
Comment (barryhughes) [inline src/Tribe/Rewrite.php]: Is this a hangover from testing?

File: src/Tribe/Rewrite.php [hunk 2]
```diff
@@ -487,11 +492,38 @@ public function remove_percent_placeholders( array $rules ) {
...
 		}
+
+		/**
+		 * A function to Add an \ before the required caracters in Regex
+		 * Avoid using it elsewhere
+		 *
+		 * @param  string $value
+		 * @return string
+		 */
+		public function _build_replace_esc_regex( $value ) {
```
Comment (barryhughes) [inline src/Tribe/Rewrite.php]: Why the `_` prefix on the method name (ditto `_esc_regex_elements()`) ... might an `@internal` doc tag be better?

File: src/Tribe/Rewrite.php [hunk 3]
```diff
@@ -487,11 +492,38 @@ public function remove_percent_placeholders( array $rules ) {
...
+		}
+
+		/**
+		 * Escapes the Required characters on a Regular Expression Word
+		 * Avoid using it elsewhere
+		 *
+		 * @param  string $base
+		 * @return string
+		 */
+		public function _esc_regex_elements( $base ) {
```
Comment (barryhughes) [inline src/Tribe/Rewrite.php]: Seems really close to what [preg_quote()](http://php.net/manual/en/function.preg-quote.php) does - is there a reason not to use that (can we add a note in the docblock if so)?

File: src/Tribe/Rewrite.php (hunk 1 above)
Comment (bordoni) [inline src/Tribe/Rewrite.php]: Yes thanks!

File: src/Tribe/Rewrite.php (hunk 3 above)
Comment (bordoni) [inline src/Tribe/Rewrite.php]: I never used `preg_quote` good catch. :wink:

**Comment by bordoni** [comment]
Thanks for the CR @barryhughes, just updated the code.

**Comment by barryhughes** [comment]
yisss

**Pull request merged**